### PR TITLE
Changes for flex cloud + use authproxy

### PR DESCRIPTION
**Pull request by pshanbhag**

### CODE REVIEW

File: pkg/client/service.go [hunk 1]
```diff
@@ -41,7 +43,7 @@ func NewComputeV2(provider *gophercloud.ProviderClient, eo *gophercloud.Endpoint
 	return compute, nil
 }
 
-// NewBlockStorageV3 creates a ServiceClient that may be used with the Cinder v3 API
```
Comment (TheNilesh) [inline pkg/client/service.go:44]: Could be a typo because func signature does not match comment

File: pkg/csi/cinder/openstack/openstack.go [hunk 2]
```diff
@@ -148,7 +149,14 @@ func InitOpenStackProvider(cfgFiles []string, httpEndpoint string) {
...
 	}
-
+	if len(cfgFiles) == 0 {
+		klog.Infof("Using internal cloud config file: %s", cfgFiles)
+		if err := createCfgFile(configFiles[0], regionName); err != nil {
+			klog.Fatalf("Failed to create cloud config file: %v", err)
+		}
+		return
+	}
+	klog.Infof("Using provided cloud config files: %s", cfgFiles)
```
Comment (TheNilesh) [inline pkg/csi/cinder/openstack/openstack.go:159]: There is no need to build `cloud.conf` in the container and no need to provide a region.

**Review by TheNilesh — CHANGES_REQUESTED** [review]
There's no need to separate Claymore from non-Claymore as long as we provide different secrets (dummy and real) to both. In my DaemonSet PR, I removed the complicated step of building `cloud.conf` from the cloud provider for this reason. I suggest we don’t add the `USE_CLAYMORE` flag.

Changes related to OpenStack v3 API look okay.

**Review by TheNilesh — APPROVED** [review]
Given the release timeline, we can handle the changes related to removing my commits separately.

File: pkg/csi/cinder/openstack/openstack.go (hunk 2 above)
Comment (pshanbhag) [reply pkg/csi/cinder/openstack/openstack.go:159]: https://platform9.atlassian.net/browse/NGPC-1057

**Comment by pshanbhag** [comment]
> There's no need to separate Claymore from non-Claymore as long as we provide different secrets (dummy and real) to both. In my DaemonSet PR, I removed the complicated step of building `cloud.conf` from the cloud provider for this reason. I suggest we don’t add the `USE_CLAYMORE` flag.
> 
> Changes related to OpenStack v3 API look okay.

I have filed a task to do the clean up in a subsequent release.

**Pull request merged**